src/nomad_catalysis/schema_packages/schema.py

```diff
@@ -0,0 +1,424 @@
+from typing import (
+    TYPE_CHECKING,
+)
+
+import numpy as np
+from ase.data import chemical_symbols
+from nomad.config import config
+from nomad.datamodel.data import ArchiveSection, EntryDataCategory, Schema
+from nomad.datamodel.metainfo.annotations import ELNAnnotation
+from nomad.datamodel.metainfo.basesections import (
+    CompositeSystem,
+    Process,
+)
+from nomad.datamodel.results import (
+    Catalyst,
+    CatalyticProperties,
+    Material,
+    Properties,
+    Results,
+)
+from nomad.metainfo import (
+    Quantity,
+    SchemaPackage,
+    Section,
+    SubSection,
+)
+from nomad.metainfo.metainfo import Category
+
+# from .catalyst_measurement import (
+#     CatalyticReactionData,
+#     CatalyticReactionData_core,
+#     Rates,
+#     ReactionConditions,
+#     ReactionConditionsSimple,
+#     ReactorSetup,
+#     add_activity,
+# )
+# from .catalyst_measurement import Product as Product_data
+# from .catalyst_measurement import Reactant as Reactant_data
+# from .catalyst_measurement import Reagent as Reagent_data
+
+if TYPE_CHECKING:
+    pass
+
+
+configuration = config.get_plugin_entry_point(
+    'nomad_catalysis.schema_packages:catalysis'
+)
+
+m_package = SchemaPackage()
+
+
+class CatalysisElnCategory(EntryDataCategory):
+    m_def = Category(label='Catalysis', categories=[EntryDataCategory])
+
+
+def add_catalyst(archive):
+    """Adds metainfo structure for catalysis data."""
+    if not archive.results:
+        archive.results = Results()
+    if not archive.results.properties:
+        archive.results.properties = Properties()
+    if not archive.results.properties.catalytic:
+        archive.results.properties.catalytic = CatalyticProperties()
+    if not archive.results.properties.catalytic.catalyst:
+        archive.results.properties.catalytic.catalyst = Catalyst()
+
+
+# helper function to retrieve nested attributes
+def get_nested_attr(obj, attr_path):
+    for attr in attr_path.split('.'):
+        obj = getattr(obj, attr, None)
+        if obj is None:
+            return None
+    return obj
+
+
+def set_nested_attr(obj, attr_path, value):
+    for attr in attr_path.split('.'):
+        obj = getattr(obj, attr, None)
+        if obj is None:
+            return
+    setattr(obj, attr[-1], value)
+
+
+def populate_catalyst_sample_info_to_results(archive, self, logger, reference=False):  # noqa: PLR0912
+    """
+    Copies the catalyst sample information (if reference provided from a reference)
```

Comment:
It should have said:"Copies the catalyst sample information into the results section, if reference=True it takes the information from the linked sample reference" I will clarify it in the code

Reply:
But what is the "linked sample reference"? Without explaining the arguments of the function it still becomes quite hard to understand. With a bit of digging in it I guess that your "self" argument can be different things? I would rather make this a class method of the sample as it ultimately acts on the sample anyways. You can then call it directly on the referenced sample in your code instead of on the parent section if that makes sense?